code/app/src/main/java/com/example/the_tarlords/MainActivity.java

```diff
@@ -261,4 +287,95 @@ private String generateNewUserId() {
         return Settings.Secure.getString(getContentResolver(), Settings.Secure.ANDROID_ID);
     }
 
+    /**
+     * Mandatory empty method here because MainActivity implements OnMapReadyCallBack
+     * the function is implemented and used in MapsFragment.java
+     * @param googleMap
+     */
+    @Override
+    public void onMapReady(@NonNull GoogleMap googleMap) {
+
+    }
+
+    /**
+     * Method to get users location and if needed the permissions
+     */
+    public void getMyLocation(String eventId) {
+        this.eventId = eventId;
+        if (ActivityCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, android.Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
+            // Permission not granted, request it
+            ActivityCompat.requestPermissions(this, new String[]{android.Manifest.permission.ACCESS_FINE_LOCATION, android.Manifest.permission.ACCESS_COARSE_LOCATION}, REQUEST_LOCATION_PERMISSION);
+
+        }else{
+            // Permission already granted, proceed to get location
+            Task<Location> task = client.getLastLocation();
+            task.addOnSuccessListener(new OnSuccessListener<Location>() {
+                @Override
+                public void onSuccess(Location location) {
+                    if (location != null){
+                        // get latitude and longitude, and user name , put it on firebase
+                        Double latitude = location.getLatitude();
+                        Double longitude = location.getLongitude();
+                        checkInLocation(latitude, longitude, eventId);
+                    }else{
+                        Log.d("maps", "users location during check-in was null");
+                    }
+
+                }
+            });
+        }
+
+    }
+    /**
+     * If location permissions are requested
+     * handle the case where they grant permission
+     */
+    @Override
+    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
+        if (requestCode == REQUEST_LOCATION_PERMISSION) {
+            // Check if the requested permissions were granted
+            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                // Permission granted, call getMyLocation() again to proceed
+                getMyLocation(eventId);
+            } else {
+                // Permission denied, we don't do anything
+                Log.d("maps", "location permissions denied");
+            }
+        }
+    }
+    /**
+     * Method to help getMyLocation put the users location onto Firebase
+     */
+    public void checkInLocation(Double lat, Double lon, String eventId){
```

Comment:
Could probably go in event with attendance (I'm also considering if  should make attendance it's own class again (probably a static class though) just since its getting a bit crowded in event , in which case it could maybe go in there)